ek9000App/src/scripts/generate.py

```diff
@@ -183,35 +196,110 @@ def write(self, fp):
                 fp.write(f'\tfield({i[0]}, "{i[1]}")\n')
             fp.write('}\n\n')
 
+
+    @staticmethod
+    def _get_raw_range(spec: SpecType) -> Tuple[int,int]:
+        """
+        Computes the raw range for the terminal
+        
+        Returns
+        -------
+        Tuple[int,int]:
+            Range interval (min,max)
+        """
+        isSigned = False
+        match spec['representation']:
+            case 'int16':
+                isSigned = True
+            case 'uint16':
+                isSigned = False
+            case other:
+                assert False
+        max = 32767
+        min = -max if spec['min'] < 0 and isSigned else 0 # Unsigned terminals still have range 0-32767
+        return (min, max)
+
+
+    @staticmethod
+    def _compute_deadbands(spec: SpecType) -> float:
+        """
+        Compute deadband parameters ADEL and MDEL based on max EGU and error specified in datasheet
+        Most terminals are rated for measurement error of <=0.3%
+        
+        Returns
+        -------
+        float:
+            The value ADEL and MDEL should be set to
+        """
+        error = spec['error'] / 100.0 # error is in %, convert that to float
+        return spec['max'] * error # Docs specify that error is relative to the full scale value
```

Comment:
This is a good approach to setting the deadband, I think. 👍
 
I like that this would reduce noisy fluctuations in a user interface (and the archiver). If users want to opt in to seeing the noise, they can always reconfigure it.